### PR TITLE
AG Grid Theme docs to site

### LINKED ISSUE #2506: Add AG Grid Theme documentation to site
AG Grid Theme is one of the last remaining Storybook component docs on the site. Once we've ported this to the new format, we can consider removing the table from the component docs index page.

```[tasklist]
### Tasks
- [x] @mikearildbrown to share a branch w/ draft MDX files based on the Storybook docs and make changes where necessary
- [ ] @libertymayc to correct any technical points/issues/changes needed
- [ ] @libertymayc to port examples to new format (if applicable)
- [ ] @mikearildbrown to send to proofreaders for final checks
- [ ] @mikearildbrown to make final copy edits and start a pull request.
```

### PR DESCRIPTION
**Pull request by libertymayc**
Adds ag grid theme docs and examples to site

closes #2506

### CODE REVIEW

**Comment by libertymayc** [comment]
#2506

**Comment by libertymayc** [comment]
@mikearildbrown PR is here pending review
did we want to add descriptions to the stories? since the theme is the entity here rather than ag grid, wasnt sure if we should be explaining which each story is since its more data grid related than this, it seems like a general overview is more suited and then just demonstrations to show the AG grid with different settings and how the theme is with those settings. example descirption would be more basic like 'AG Grid shown with checkboxes'.
thoughts?

File: site/docs/components/ag-grid-theme/examples.mdx [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+### Custom Filter
+
+description
+
+</LivePreview>
+<LivePreview componentName="ag-grid-theme" exampleName="Default" >
+
+### Default
+
+description
```
[the remark this answers is not in the document]
Comment (origami-z) [reply site/docs/components/ag-grid-theme/examples.mdx:58]:
```suggestion
Default look when applying Salt theme
```

File: site/docs/components/ag-grid-theme/examples.mdx [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+### Column spanning
+
+Rows can span multiple columns.
+
+</LivePreview>
+<LivePreview componentName="ag-grid-theme" exampleName="ContextMenu" >
+
+### Context menu
+
+description
```
Comment (libertymayc) [inline site/docs/components/ag-grid-theme/examples.mdx:44]: The menu can be opened on the column to view details or filters.

File: site/docs/components/ag-grid-theme/examples.mdx [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+  $ref: ./#/data
+---
+
+<AGThemeProvider>
+<LivePreviewControls>
+<LivePreview componentName="ag-grid-theme" exampleName="CheckboxSelection" >
+
+### Checkbox selection
+
+description
```
Comment (libertymayc) [inline site/docs/components/ag-grid-theme/examples.mdx:16]:
```suggestion
Rows can be selected via checkboxes.
```

File: site/docs/components/ag-grid-theme/examples.mdx [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+</LivePreview>
+<LivePreview componentName="ag-grid-theme" exampleName="Coloration" >
+
+### Coloration
+
+description
```
Comment (libertymayc) [inline site/docs/components/ag-grid-theme/examples.mdx:23]:
```suggestion
Cells, rows or columns can have color applied to them.
```

File: site/docs/components/ag-grid-theme/examples.mdx [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+### Floating filter
+
+The filter remains visible as the user scrolls.
+
+</LivePreview>
+<LivePreview componentName="ag-grid-theme" exampleName="HDCompact" >
+
+### HD Compact
+
+description
```
Comment (libertymayc) [inline site/docs/components/ag-grid-theme/examples.mdx:86]: HD Compact theme is available by applying the `ag-grid-salt-high-compact` classname.

File: site/docs/components/ag-grid-theme/examples.mdx [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+<LivePreview componentName="ag-grid-theme" exampleName="FloatingFilter" >
+
+### Floating filter
+
+The filter remains visible as the user scrolls.
+
+</LivePreview>
+<LivePreview componentName="ag-grid-theme" exampleName="HDCompact" >
+
+### HD Compact
```
Comment (libertymayc) [inline site/docs/components/ag-grid-theme/examples.mdx:84]: @origami-z should this theme created for stakeholder  be exposed?

Comment (origami-z) [reply site/docs/components/ag-grid-theme/examples.mdx:84]: This is fine, this will receive first class support, would be good to be documented

**Review by origami-z — COMMENTED** [review]
The build probably breaks due to type error, `cd site && yarn build` has error around `.png` imports. Can try put in `public` then reference directly via path

**Review by joshwooding — CHANGES_REQUESTED** [review]
At a quick look, the examples don't work great with the playground. The HD compact example needs some work, so does the "no data overlay" one and the pinned rows example doesn't work

**Comment by libertymayc** [comment]
> At a quick look, the examples don't work great with the playground. The HD compact example needs some work, so does the "no data overlay" one and the pinned rows example doesn't work

can this work be passed over to @alycrys ?

**Review by joshwooding — DISMISSED** [review]
Nice work!

**Review by Fercas123 — DISMISSED** [review]

**Comment by origami-z** [comment]
Found a typo in usage page, updated

**Review by Fercas123 — APPROVED** [review]

**Review by joshwooding — APPROVED** [review]

**Pull request merged**